src/mlpack/methods/ann/loss_functions/poisson_nll_loss_impl.hpp

```diff
@@ -70,7 +73,10 @@ void PoissonNLLLoss<InputDataType, OutputDataType>::Backward(
   if (logInput)
     output = (arma::exp(input) - target);
   else
+  {
+    CheckProbs(input);
```

Comment:
Same comment here. And if I am not mistaken the same input has already been checked inside `Forward()`. I mean the backward pass is being applied after the forward pass.

Reply:
Yes. We can remove it from `Backward()`. :+1: